BlueprintUI/Sources/Layout/Decorate.swift

```diff
@@ -0,0 +1,236 @@
+        /// `horizontal` amount on the left and right, and the `vertical` amount on the top and bottom.
+        public static func inset(horizontal : CGFloat = 0.0, vertical : CGFloat = 0.0) -> Self {
+            .inset(UIEdgeInsets(top: vertical, left: horizontal, bottom: vertical, right: horizontal))
+        }
+        
+        /// The decoration element is positioned in the given corner of the
+        /// content element, optionally offset by the provided amount.
+        case corner(Corner, UIOffset = .init())
+        
+        /// Allows you to provide custom positioning for the decoration, based on the passed context.
+        case custom((CustomContext) -> CGRect)
+        
+        /// Information provided to the `.custom` positioning type.
+        public struct CustomContext {
+            
+            /// The frame of the content element within the `Decorate` element.
+            public var contentFrame : CGRect
+            
+            /// The environment the element is being rendered in.
+            public var environment : Environment
+        }
+        
+        func frame(
+            with contentFrame : CGRect,
+            decoration : Element,
+            environment : Environment
+        ) -> CGRect {
+            
+            switch self {
+            case .inset(let inset):
+                return contentFrame.inset(by: inset.negated)
+                
+            case .corner(let corner, let offset):
+                
+                let size = decoration.content.measure(in: .init(contentFrame.size), environment: environment)
+                
+                let center : CGPoint = {
+                    switch corner {
+                    case .topLeft:
+                        return .zero
+                    case .topRight:
+                        return CGPoint(x: contentFrame.maxX, y: 0)
+                    case .bottomRight:
+                        return CGPoint(x: contentFrame.maxX, y: contentFrame.maxY)
+                    case .bottomLeft:
+                        return CGPoint(x: 0, y: contentFrame.maxY)
+                    }
+                }()
+                
+                return CGRect(
+                    origin: CGPoint(
+                        x: center.x - (size.width / 2.0),
+                        y: center.y - (size.height / 2.0)
+                    ),
+                    size: size
+                ).offset(
+                    by: CGPoint(x: offset.horizontal, y: offset.vertical)
+                )
+                
+            case .custom(let provider):
+                return provider(.init(contentFrame: contentFrame, environment: environment))
+            }
+        }
+    }
+}
+
+
+extension Element {
+    
+    /// Places a decoration element behind or in front of the given `wrapped` element,
+    /// and positions it according to the `position` parameter.
+    ///
+    /// See the `Decorate` element for more.
+    ///
+    public func decorate(
```

Comment:
Can we expose this with [background] and [overlay] modifiers instead (maybe renaming the latter to disambiguate from the `Overlay` element)?

[background]: https://developer.apple.com/documentation/swiftui/view/background(_:alignment:)
[overlay]: https://developer.apple.com/documentation/swiftui/view/overlay(_:alignment:)

Reply:
I went with this because of potential confusion – eg background sounds like I'm giving a background to my current element, and overlay is confused with `Overlay` as you mention.